### PR TITLE
add esync macro

### PR DESCRIPTION
**Pull request by aureocarneiro**
credits to @CarlaCTaka

add icepap esync macro

### CODE REVIEW

**Comment by aureocarneiro** [comment]
I see bumpversion also push a tag, so I guess its done on `master` branch.

**Comment by rhomspuron** [comment]
Hi @aureocarneiro and @CarlaCTaka,

Thanks! I will integrate it. 

On the last version we implement the automatic esync command before any movement, because we detect some problems with the trigger generation.

File: sardana_icepap/macro/esync.py [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+
+    Args:
+            macro (Macro): running macro
+            motor (Type.Motor): motor device
+    """
+    controllers = macro.getControllers()
+    ctrl_name = motor.controller
+    controller = controllers[ctrl_name]
+    return isIcepapController(controller)
+
```
Comment (rhomspuron) [inline sardana_icepap/macro/esync.py]: You can remove it, it is defined on utils module

File: sardana_icepap/macro/esync.py [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+    """
+    Checks if pool controller is of type IcepapController
+
+    Args:
+            controller (Type.Controller): controller device
+    """
+    controller_class_name = controller.getClassName()
+    return controller_class_name == "IcepapController"
+
+
```
Comment (rhomspuron) [inline sardana_icepap/macro/esync.py]: It is defined on utils.py

**Review by rhomspuron — CHANGES_REQUESTED** [review]

File: sardana_icepap/macro/esync.py (hunk 1 above)
Comment (aureocarneiro) [reply sardana_icepap/macro/esync.py]: done

File: sardana_icepap/macro/esync.py (hunk 2 above)
Comment (aureocarneiro) [reply sardana_icepap/macro/esync.py]: done

**Review by rhomspuron — APPROVED** [review]

**Pull request merged**